### PR TITLE
election(engine): add retry for release

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/7168

### What is changed and how it works?

Add `updateRecord` to retry updating the record in an optimistic way. 
There is public method is changed, so no new tests are added.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/pkg/election/elector.go [hunk 1]
```diff
@@ -247,31 +220,55 @@ func (e *Elector) release(ctx context.Context, removeSelf bool) error {
...
-	record, err := s.Get(ctx)
-	if err != nil {
-		return errors.Trace(err)
+func (e *Elector) updateRecord(ctx context.Context, f func(*Record) error) error {
+	// Divide 2 is for more retries.
+	backoffBaseDelayInMs := int64(e.config.RenewInterval/time.Millisecond) / 2
+	// Make sure the retry delay is less than the deadline, otherwise the retry has no chance to execute.
+	backoffMaxDelayInMs := int64(e.config.RenewDeadline/time.Millisecond) / 2
+	if deadline, ok := ctx.Deadline(); ok {
+		backoffMaxDelayInMs = int64(deadline.Sub(time.Now())/time.Millisecond) / 2
```
Comment (amyangfei) [inline engine/pkg/election/elector.go:245]: Should use the smaller one for `backoffMaxDelayInMs`

File: engine/pkg/election/elector.go [hunk 2]
```diff
@@ -247,31 +220,55 @@ func (e *Elector) release(ctx context.Context, removeSelf bool) error {
...
+
+		if err := s.Update(ctx, record); err != nil {
+			return errors.Trace(err)
+		}
+		e.setObservedRecord(record)
+		return nil
+	}, retry.WithBackoffBaseDelay(backoffBaseDelayInMs),
+		retry.WithBackoffMaxDelay(backoffMaxDelayInMs),
+		retry.WithIsRetryableErr(func(err error) bool {
+			log.Debug("failed to update record", zap.Error(err))
```
Comment (amyangfei) [inline engine/pkg/election/elector.go:268]: There is a info log before, `log.Info("renew lease failed, retry after random delay",`
Is it better to use Info level here

File: engine/pkg/election/elector.go (hunk 1 above)
Comment (gozssky) [reply engine/pkg/election/elector.go]: Addressed in https://github.com/pingcap/tiflow/pull/7338/commits/58fa73cdf26b71098210e80fe8d4212227b53cd4.

File: engine/pkg/election/elector.go (hunk 2 above)
Comment (gozssky) [reply engine/pkg/election/elector.go]: Addressed. Info for conflict and Warn for other errors.

**Review by amyangfei — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**